engine/src/main/java/org/terasology/world/ChunkView.java

```diff
@@ -185,6 +185,46 @@
      * @param newState
      */
     void setLiquid(int x, int y, int z, LiquidData newState);
+    
+    /**
+     * Gets one of the per-block custom data values at the given position. Returns 0 outside the view.
+     *
+     * @param index The index of the extra data field
+     * @param x
+     * @param y
+     * @param z
+     * @return The (index)th extra-data value at the given position
+     */
+    int getExtraData(int index, int x, int y, int z);
```

Comment:
The main reason I included this was just consistency with the other methods. Looking at the engine at least, it does seem that the (int, int, int) form is used somewhat less than the Vector3i form (for the (get/set)Block methods), but in some cases using 3 separate coordinates was apparently found more convenient, so I feel it may as well be permitted here, too. It involves a tiny bit less indirection, and the error of providing arguments in the wrong order should generally be quite easy to trace back to its origin anyway.